### ISSUE TITLE
Dark Theme

### ISSUE THREAD

**Issue by CharliePearce** [agersant/polaris#54]
Going with the trend of modern applications nowadays, would it be possible to implement a dark theme to Polaris as a toggle?

I've hacked together a basic CSS override that works once a user is signed in (might need tweaking etc. for accessibility).

```
.paneHeader, .paneContent, player, router route auth {
    background-color: rgb(27, 38, 44);
}

.paneContent * {
    color: rgb(255, 255, 255);
}

.paneHeader * {
    color: rgb(255, 255, 255);
}

breadcrumbs ul li, .paneContent button, .paneHeader button {
    background-color: rgb(15, 76, 117);
    color: rgb(255, 255, 255);
}

.paneHeader select, .paneHeader input, .paneContent select, .paneContent input {
    color: rgb(0, 0, 0);
}

button {
    border: 0 !important;
}

playlist tr:nth-child(2n) {
    background-color: rgb(35, 46, 52);
}

player * {
    color: rgb(255, 255, 255) !important;
}

router route settings div, router route browser div, router route playlists div, playlist, playlist div, router route player {
    border-color: rgb(35, 46, 52) !important;
}

menu .selected {
    background-color: rgb(15, 76, 117);
}
```

**Comment by agersant** [comment]
I think this is a good idea, I can't promise when I'll get to it but this is definitely something I want to implement.

**Comment by agersant** [comment]
Turns out I did have a bit of time to start working on this!

What do you think of this prototype?
![image](https://user-images.githubusercontent.com/817256/71965961-8cd78e80-31b5-11ea-83bd-c8f0ccfe8bbc.png)

**Comment by CharliePearce** [comment]
I think the white on blue contrast is a bit difficult to read on the thinner text in the breadcrumbs, but otherwise is just what I was looking for in a dark theme.

Keeping the highlight consistent across both is ideal, so it may be worth increasing the font weight in the breadcrumbs if possible (600 seems OK) or alternately in the dark theme the highlight could be dimmed down to increase the contrast between the text & bg.

The second option also makes the UI WCAG compliant which is always a bonus :).

```
breadcrumbs ul li {
    font-weight: 600;
}
```
![image](https://user-images.githubusercontent.com/2099655/71966864-448e8f80-31fa-11ea-8460-e14253e7afbc.png)

```
breadcrumbs ul li {
    background-color: #095d78;
}

menu .selected {
    background-color: #095d78;
}

playlist tr.nowPlaying td, playlist tr.nowPlaying td * {
    background-color: #095d78;
}

playlist tr.nowPlaying td.song .trackArtist {
    color: #DDD;
}
```
![image](https://user-images.githubusercontent.com/2099655/71967389-355c1180-31fb-11ea-9300-00542b8f5397.png)

**Comment by agersant** [comment]
Thanks! I agree about the font weight, I think that would also benefit the light theme so I'll change it.

I think I'll make several variants of the accent color, including lower contrast one like your second image.

**Comment by agersant** [comment]
I merged the theming work to master, this was fun to iterate on. I ended up shipping 4 theme bases (Light/Dark/Blue/Brown) and a color picker for the accent color.

The settings are saved for each user in the Polaris DB, and adding new theme bases is very easy (https://github.com/agersant/polaris-web/commit/9e9dd56b0ac505230c3894161b7eded4063a72d3).

Here is a screenshot from each base:

![image](https://user-images.githubusercontent.com/817256/72202836-e986b380-3418-11ea-8840-11886478f28e.png)


![image](https://user-images.githubusercontent.com/817256/72202817-c825c780-3418-11ea-852e-5b00e6dba4c7.png)

![image](https://user-images.githubusercontent.com/817256/72202810-bcd29c00-3418-11ea-8f59-4177ce3138b7.png)

If you are setup to run Polaris from latest, I would love to hear some feedback before this makes it into a release.
